### DISCUSSION TITLE
How to list all pubs retrieved from search_pubs?

### DISCUSSION THREAD

**Discussion by stanleyrhodes** [scholarly-python-package/scholarly#286] [Q&A] [answered]
I used `search_query = scholarly.search_pubs('"hierarchy they mean" AND social')` and `search_query.total_results` gives 4, which is correct, there are 4 results. `scholarly.pprint(next(search_query))` only gives me one result. However, how do I list the 4 results?

The documentation seems to give repeated examples of only returning one pub's information. But I'm trying to use scholarly to assist in a very large literature review--hundreds of results for specific searches. I was using Publish or Perish for this, but with Google Scholar it only returns the truncated results, whereas Scholarly seems to dig into the javascript control which returns the untruncated results (journal title and authors).

Any help would be appreciated. I'm very interested in whether Scholarly can be a powerful tool for large literature reviews.

**Comment by ipeirotis** [comment, marked as answer]
> I used `search_query = scholarly.search_pubs('"hierarchy they mean" AND social')` and `search_query.total_results` gives 4, which is correct, there are 4 results. `scholarly.pprint(next(search_query))` only gives me one result. However, how do I list the 4 results?
> 
> 
> 
> The documentation seems to give repeated examples of only returning one pub's information. But I'm trying to use scholarly to assist in a very large literature review--hundreds of results for specific searches. I was using Publish or Perish for this, but with Google Scholar it only returns the truncated results, whereas Scholarly seems to dig into the javascript control which returns the untruncated results (journal title and authors).
> 
> 
> 
> Any help would be appreciated. I'm very interested in whether Scholarly can be a powerful tool for large literature reviews.

The command next in Python returns the next element of the iterator. 

Just iterate using for (instead of next) and you will be able to go through the full list of elements.

**Reply by stanleyrhodes** [reply]
Thank you. I had tried that and it didn't work, but I see now that this iterator must be special compared to the iterators I'm used to dealing with. It seems to expire when the last item is called, when StopIteration happens. So I had already expired the iterator and thus my for loop wasn't working. If that's nonsensical and wasn't my problem, then I have no idea what I did. I'll need to dig into this more, I'm barely above a white belt with Python.

```
sq_array = []
for i in search_query:
    sq_array.append(i)
```

lets me have a list to work with. In case anyone passing through is wondering how `for` would be used in this case.

**Reply by ipeirotis** [reply]
The "for i in search_query" is effectively doing i=next(search_query) until the next command returns an exception.
